### ISSUE TITLE
[CUDA] Lowering issue with torch.aten.adaptive_max_pool1d

### ISSUE THREAD

**Issue by Cerulime** [iree-org/iree#23213]
### What happened?

In this MLIR:

```MLIR
module @module {
  func.func @main(%arg0: !torch.vtensor<[270,270,270],f32>) -> !torch.vtensor<[270,270,31281],f32> attributes {torch.assume_strict_symbolic_shapes} {
    %int31281 = torch.constant.int 31281
    %0 = torch.prim.ListConstruct %int31281 : (!torch.int) -> !torch.list<int>
    %result0, %result1 = torch.aten.adaptive_max_pool1d %arg0, %0 : !torch.vtensor<[270,270,270],f32>, !torch.list<int> -> !torch.vtensor<[270,270,31281],f32>, !torch.vtensor<[270,270,31281],si64>
    return %result0 : !torch.vtensor<[270,270,31281],f32>
  }
}
```

The generated LLVM IR:

```
; Function Attrs: alwaysinline mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite, inaccessiblemem: write)
define ptx_kernel void @main_async_dispatch_1_reduction_2280384900x2_i1xf32(ptr addrspace(1) noalias noundef nonnull readonly align 16 captures(none) %0, ptr addrspace(1) noalias noundef nonnull writeonly align 16 captures(none) %1) local_unnamed_addr #2 {
  %3 = tail call range(i32 0, 32) i32 @llvm.nvvm.read.ptx.sreg.tid.x()
  call void @llvm.assume(i1 true) [ "align"(ptr addrspace(1) %0, i64 64) ]
  call void @llvm.assume(i1 true) [ "align"(ptr addrspace(1) %1, i64 64) ]
  %4 = tail call range(i32 0, 71262029) i32 @llvm.nvvm.read.ptx.sreg.ctaid.x()
  %5 = shl nuw i32 %4, 5
  %6 = sub nuw i32 -2014582396, %5
  %7 = icmp sgt i32 %6, %3
  br i1 %7, label %.lr.ph.preheader, label %._crit_edge

.lr.ph.preheader:                                 ; preds = %2
  %8 = or disjoint i32 %3, %5
  %9 = zext i32 %8 to i64
  %10 = sdiv i32 %8, 31281
  %11 = mul i32 %10, 31281
  %.decomposed = sub i32 %8, %11
  %12 = udiv i32 %10, 270
  %13 = mul i32 %12, 270
  %.decomposed2 = sub i32 %10, %13
  %14 = zext nneg i32 %12 to i64
  %15 = mul nsw i32 %.decomposed, 270
  %16 = udiv i32 %15, 31281
  %17 = add nsw i32 %15, 269
  %18 = udiv i32 %17, 31281
  %19 = getelementptr float, ptr addrspace(1) %1, i64 %9
  %narrow = mul nuw nsw i32 %.decomposed2, 271
  %20 = zext nneg i32 %narrow to i64
  %.idx = mul nuw nsw i64 %14, 292680
  %21 = getelementptr inbounds nuw i8, ptr addrspace(1) %0, i64 %.idx
  %22 = getelementptr inbounds nuw float, ptr addrspace(1) %21, i64 %20
  %23 = zext nneg i32 %16 to i64
  %24 = getelementptr inbounds nuw float, ptr addrspace(1) %22, i64 %23
  %25 = load float, ptr addrspace(1) %24, align 4
  %.not = icmp samesign ugt i32 %16, %18
  %26 = select i1 %.not, float 0xFFF0000000000000, float %25
  %27 = fcmp ogt float %26, 0xFFF0000000000000
  %28 = select i1 %27, float %26, float 0xFFF0000000000000
  %29 = zext nneg i32 %16 to i64
  %30 = getelementptr inbounds nuw float, ptr addrspace(1) %22, i64 %29
  %31 = getelementptr inbounds nuw i8, ptr addrspace(1) %30, i64 4
  %32 = load float, ptr addrspace(1) %31, align 4
  %.not.1.not = icmp samesign ult i32 %16, %18
  %33 = select i1 %.not.1.not, float %32, float 0xFFF0000000000000
  %34 = fcmp ogt float %33, %28
  %35 = select i1 %34, float %33, float %28
  store float %35, ptr addrspace(1) %19, align 4
  br label %._crit_edge

._crit_edge:                                      ; preds = %.lr.ph.preheader, %2
  ret void
}
```

`%8` was `i32` and it tried to get the Global Thread ID. But the input was larger than MAX_INT32. Then `%10` used `sdiv` and generated a negative number. Casted by `%12` and `udiv`, the index became very large.

The result was below, detected by computer-sanitizer:

```
========= COMPUTE-SANITIZER
========= Invalid __global__ read of size 4 bytes
=========     at main_async_dispatch_1_reduction_2280384900x2_i1xf32+0x180
=========     by thread (0,0,0) in block (67109200,0,0)
=========     Address 0x84360798db24 is out of bounds
=========     and is 4,632,706,865,445 bytes after the nearest allocation at 0x7fff64c00000 of size 512 bytes
=========     Saved host backtrace up to driver entry point at kernel launch time
=========         Host Frame: cuLaunchKernel [0x39f4e4] in libcuda.so.1
=========         Host Frame: iree_hal_cuda_stream_command_buffer_dispatch [0x134f98] in _runtime.cpython-312-x86_64-linux-gnu.so
=========         Host Frame: iree_hal_deferred_command_buffer_apply_dispatch [0x15ffb3] in _runtime.cpython-312-x86_64-linux-gnu.so
=========         Host Frame: iree_hal_deferred_command_buffer_apply [0x15f475] in _runtime.cpython-312-x86_64-linux-gnu.so
=========         Host Frame: iree_hal_deferred_work_queue_worker_execute [0x138607] in _runtime.cpython-312-x86_64-linux-gnu.so
=========         Host Frame: iree_thread_start_routine [0x1664da] in _runtime.cpython-312-x86_64-linux-gnu.so
=========         Host Frame: start_thread in pthread_create.c:447 [0x9caa3] in libc.so.6
=========         Host Frame: clone3 in clone3.S:78 [0x129c6b] in libc.so.6
```

The block number `67109200` multiplied by `32` was `2,147,494,400`. So it should use `i64` or `u32`.

### Steps to reproduce your issue

Running this code with compute-sanitizer:

```python
import iree.turbine.aot as aot
import iree.compiler.tools as iree_tools
import iree.runtime as ireert
import os
import torch
import torch.nn as nn

dump_dir = os.path.join(os.getcwd(), "iree_cuda_dumps")
os.makedirs(dump_dir, exist_ok=True)
mlir_filename = "AdapMax1d.mlir"
vmfb_filename = "AdapMax1d.vmfb"

class Model1(nn.Module):
    def __init__(self):
        super().__init__()
        self.pool = nn.AdaptiveMaxPool1d(output_size=31281) # 31281 exactly triggers the issue

    def forward(self, x):
        return self.pool(x)

model = Model1()
example_input = torch.randn(270, 270, 270)
export_output = aot.export(model, example_input)
export_output.save_mlir(os.path.join(dump_dir, mlir_filename))

try:
    iree_tools.compile_file(
        input_file=os.path.join(dump_dir, mlir_filename),
        output_file=os.path.join(dump_dir, vmfb_filename),
        target_backends=["cuda"],
        extra_args=[
            f"--iree-hal-dump-executable-intermediates-to={dump_dir}",
            "--iree-stream-emulate-memset",
            "--iree-flow-zero-fill-empty-tensors",
            "--iree-cuda-target=sm_86"
        ]
    )
    print("Compilation successful.")
    print("Running inference on IREE (CUDA backend)...")
    
    ireert.flags.parse_flags("--cuda_async_allocations=false --cuda_use_streams=false")
    config = ireert.Config("cuda") 
    ctx = ireert.SystemContext(config=config)
    vm_module = ireert.VmModule.mmap(ctx.instance, os.path.join(dump_dir, vmfb_filename))
    ctx.add_vm_module(vm_module)
    iree_func = ctx.modules.module.main
    input_numpy = example_input.detach().cpu().numpy()
    result = iree_func(input_numpy)
    
    print(f"\nInput shape: {example_input.shape}")
    print(f"Output shape (IREE): {result.to_host().shape}")

except Exception as e:
    print(f"Error: {e}")

```

### Version information

commit hash: 7d91727df3

Using Python 3.12.3 environment at: iree_venv
Package            Version                                              
------------------ ----------------------------------------------------
certifi            2026.1.4
charset-normalizer 3.4.4
decorator          5.2.1
filelock           3.20.0
fsspec             2025.12.0
idna               3.11
iree-base-compiler 3.10.0.dev0+7d91727df3c9013b19c9417f5a9fd060d60c05cb
iree-base-runtime  3.10.0.dev0+7d91727df3c9013b19c9417f5a9fd060d60c05cb
iree-turbine       3.9.0
jinja2             3.1.6
markupsafe         2.1.5
ml-dtypes          0.5.4
mpmath             1.3.0
networkx           3.6.1
numpy              2.4.1
pip                25.3
requests           2.32.5
setuptools         80.9.0
sympy              1.14.0
torch              2.9.1+cpu
typing-extensions  4.15.0
urllib3            2.6.3
wheel              0.45.1

### Additional context

This bug was found by a fuzzer. The related issue is: [https://github.com/pytorch/pytorch/issues/145453](url). I am new to this community, if I did anything wrong, please let me know...
